repos/fdbt-site/src/pages/api/deleteCap.ts

```diff
@@ -14,6 +18,30 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
         }
 
         const noc = getAndValidateNoc(req, res);
+        const cap = await getCapByNocAndId(noc, id);
+
+        if (!cap) {
+            throw new Error('unable to find cap with provided id');
+        }
+
+        const products = await getAllProductsByNoc(noc);
+        const matchingJsonLinks = products.map((product) => product.matchingJsonLink);
+        const tickets: TicketWithIds[] = await Promise.all(
+            matchingJsonLinks.map(async (link) => {
+                return await getProductsMatchingJson(link);
+            }),
+        );
+
+        const productsUsingCap = tickets.filter((ticket) => 'cap' in ticket && ticket.cap && ticket.cap.id === id);
```

Comment:
```suggestion
        const productsUsingCap = tickets.filter((ticket) => ticket.cap && ticket.cap.id === id);
```

Reply:
<img width="1728" alt="image" src="https://user-images.githubusercontent.com/112861163/219038020-bc6eafe2-d562-4d1d-ae08-8dece828f69d.png">

Reply:
OK no worries - it should work but our types are a bit of a mess.